presto-main/src/main/java/com/facebook/presto/sql/planner/planPrinter/RowExpressionFormatter.java

```diff
@@ -85,20 +85,25 @@ else if (standardFunctionResolution.isBetweenFunction(node.getFunctionHandle()))
             }
             else if (standardFunctionResolution.isLikeFunction(node.getFunctionHandle())) {
                 RowExpression value = node.getArguments().get(0);
-                CallExpression patternCallExpression = (CallExpression) node.getArguments().get(1);
-
-                // second LIKE argument is:
-                //  CAST(pattern as LikePattern), if escape is not present
-                //  LIKE_PATTERN(pattern, escape), if escape is present
-                if (standardFunctionResolution.isCastFunction(patternCallExpression.getFunctionHandle())) {
-                    RowExpression pattern = patternCallExpression.getArguments().get(0);
-                    return String.format("%s LIKE %s", formatRowExpression(session, value), formatRowExpression(session, pattern));
-                }
-                else {
-                    RowExpression pattern = patternCallExpression.getArguments().get(0);
-                    RowExpression escape = patternCallExpression.getArguments().get(1);
-                    return String.format("%s LIKE %s ESCAPE %s", formatRowExpression(session, value), formatRowExpression(session, pattern), formatRowExpression(session, escape));
+                RowExpression patternRowExpression = node.getArguments().get(1);
+
+                if (patternRowExpression instanceof CallExpression) {
+                    CallExpression patternCallExpression = (CallExpression) patternRowExpression;
+                    // second LIKE argument is:
+                    //  CAST(pattern as LikePattern), if escape is not present
+                    //  LIKE_PATTERN(pattern, escape), if escape is present
+                    if (standardFunctionResolution.isCastFunction(patternCallExpression.getFunctionHandle())) {
+                        RowExpression pattern = patternCallExpression.getArguments().get(0);
+                        return String.format("%s LIKE %s", formatRowExpression(session, value), formatRowExpression(session, pattern));
+                    }
+                    else if (standardFunctionResolution.isLikePatternFunction(patternCallExpression.getFunctionHandle())) {
```

Comment:
Does this belong inside the if? Looks like it's unrelated

Reply:
Yes, those are created [here](https://github.com/prestodb/presto/blob/79d8a59d5a42baeb41ac3f21c3c9c7323782e30d/presto-main/src/main/java/com/facebook/presto/sql/relational/SqlToRowExpressionTranslator.java#L735-L757).